app/repository/OrderRepository.scala

```diff
@@ -2,13 +2,36 @@ package repository
 
 import com.google.inject.Inject
 import dao.util.ShippearDAO
-import model.internal.OrderState.CANCELLED
+import model.internal.OrderState.{CANCELLED, PENDING_PICKUP}
+import model.internal.UserType._
 import model.internal.{Order, User}
 import service.Exception.NotFoundException
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class OrderRepository @Inject()(userRepository: UserRepository)(implicit ec: ExecutionContext) extends ShippearRepository[Order] {
+  def assignCarrier(orderId : String, carrierId : String, qrCode : Array[Byte]): Future[Unit] ={
+    for{
+      order <- super.findOneById(orderId)
+      newOrder = order.copy(carrierId = Some(carrierId), qrCode = Some(qrCode), state = PENDING_PICKUP)
+      _ = update(newOrder)
+      _ = userRepository.updateUserOrder(newOrder.applicantId, newOrder)
+      _ = userRepository.updateUserOrder(newOrder.participantId, newOrder)
+      _ = userRepository.updateUserOrder(carrierId, newOrder)
```

Comment:
Aca esta bien, lo que haria es utilizar en todos los lugares la flechita `<-` en vez de el `=`.

Lo que hace esto es que, si algun update falla, el `for` corta su ejecucion y lanza el error. De esta manera podriamos saber en que parte fallo.

Con el `=` en cambio, corre todo en paralelo y si alguno de esos updates falla, nunca nos enterariamos